base/reducedim.jl

```diff
@@ -325,12 +325,18 @@ function findminmax!{T,N}(f, Rval, Rind, A::AbstractArray{T,N})
     Rval, Rind
 end
 
-# findmin
+
+"""
+    findmin!(rval, rind, A, [init=true]) -> (minval, index)
+
+Find the minimum of `A` and the corresponding linear index along singleton
+dimensions of `rval` and `rind`, and store the results in `rval` and `rind`.
+"""
 function findmin!{R}(rval::AbstractArray{R},
                      rind::AbstractArray,
                      A::AbstractArray;
                      init::Bool=true)
-    findminmax!(LessFun(), initarray!(rval, typemax(R), init), rind, A)
+    findminmax!(LessFun(), initarray!(rval, MinFun(), init), rind, A)
```

Comment:
@tkelman the bug was here.

the `initarray!` function expects one of the `Functor`s, not a number of type `R`.